generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -63,18 +63,24 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
                                    ShadingLanguageVersion shadingLanguageVersion) {
     this.generator = generator;
     this.generationParams = generationParams;
-    // TODO: there are many more identities that we can easily play with here, e.g. bitwise and 1
-    // for integer types
     this.expressionIdentities = new ArrayList<>();
     this.shadingLanguageVersion = shadingLanguageVersion;
 
+    // TODO: there are many more identities that we can easily play with here
     expressionIdentities.add(new IdentityAddSubZero());
     expressionIdentities.add(new IdentityMulDivOne());
     expressionIdentities.add(new IdentityAndTrue());
     expressionIdentities.add(new IdentityOrFalse());
-    expressionIdentities.add(new IdentityNotNot());
+    expressionIdentities.add(new IdentityLogicalNotNot());
     expressionIdentities.add(new IdentityTernary());
 
+    if (shadingLanguageVersion.supportedBitwiseOperations()) {
```

Comment:
These look good!

How about also adding:

- IdentityBitwiseOrZero // e -> e | 0

And possibly change "IdentityBitwiseOr" to "IdentityBitwiseOrSelf" to help disambiguate the two.

I did also wonder about adding:

- IdentityBitwiseAndOnes // e -> e & 0xFFFFFFFF, but actually it's not clear to me whether 0xFFFFFFFF is always well-defined in GLSL due to the limited range of integers in some contexts.

Reply:
Done.